components/motor/gpio/basic.go

```diff
@@ -93,6 +94,7 @@ var _ = motor.LocalMotor(&Motor{})
 
 // A Motor is a GPIO based Motor that resides on a GPIO Board.
 type Motor struct {
+	mu                       sync.Mutex
```

Comment:
request: can you please move the 2 state variables (on and powerPct) away from the config fields and drop a comment above? we have this patter on other drivers.


e.g.
```
// config
....

// state
on bool
powerPct float64
```